### ISSUE TITLE
REPL let bindings can't define infix operators

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#759]
I can define named values or functions at the REPL with `let`, but I can't define infix operators:
```
Cryptol> let f x y = x && y
Cryptol> let x & y = x && y

Parse error at <interactive>:1:7,
  unexpected: &
Cryptol> let (&) = (&&)

Parse error at <interactive>:1:6,
  unexpected: &
Cryptol> let (&) x y = x && y

Parse error at <interactive>:1:6,
  unexpected: &
```
I would expect all of those forms to work.

**Comment by robdockins** [comment]
REPL type synonyms would also be convenient

**Comment by robdockins** [comment]
Fixed via #1255
